File: src/decoders/password-rules.ts
import type { PasswordRules } from "~/models";
import { decodeDomain } from "./domain";

export const decodePasswordRules = (rules: any): PasswordRules => {
  return {
    maxLength: rules.max,
    minLength: rules.min,
    rules: decodeDomain(rules.regles.V)
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/password-rules.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.PasswordRules
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.int
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

fun decodePasswordRules(rules: JsonObject) =
    PasswordRules(
        maxLength = rules["max"]!!.jsonPrimitive.int,
        minLength = rules["min"]!!.jsonPrimitive.int,
        rules = decodeDomain(rules["regles"]!!.jsonObject["V"]!!.jsonPrimitive.content))
